nautobot/extras/datasources/registry.py

```diff
@@ -48,20 +45,26 @@ def refresh_datasource_content(model_name, record, request, job_result, delete=F
                     job_result.log(
                         f"Error while refreshing {entry.name}: {exc}", level_choice=LogLevelChoices.LOG_FAILURE
                     )
-                    job_result.set_status(JobResultStatusChoices.STATUS_FAILURE)
-                job_result.save()
-            job_result.log(f"Data refresh from {record} complete!", level_choice=LogLevelChoices.LOG_INFO)
-            job_result.save()
+                    raise
     else:
-        # TODO(jathan): Review how we can eliminate these deliberate `set_status()` and `save()`
-        # calls,moving them to Celery-friendly primitives instead.
         for entry in get_datasource_contents(model_name):
```

Comment:
Should be moved to a helper method so we don't have to maintain 2 copies of the same code (see comment above)